runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -635,6 +635,29 @@ local function add_injection(t, tree_index, pattern, lang, combined, ranges)
   table.insert(t[tree_index][lang][pattern].regions, ranges)
 end
 
+-- TODO(clason): replace by refactored `ts.has_parser` API (without registering)
+---@param lang string parser name
+---@return boolean # true if parser for {lang} exists on rtp
+local has_parser = function(lang)
+  return vim._ts_has_language(lang)
+    or #vim.api.nvim_get_runtime_file('parser/' .. lang .. '.*', false) > 0
+end
+
+--- Return parser name for language (if exists) or filetype (if registered and exists)
+---
+---@param alias string language or filetype name
+---@return string? # resolved parser name
+local resolve_lang = function(alias)
+  if has_parser(alias) then
+    return alias
+  else
+    local lang = vim.treesitter.language.get_lang(alias)
+    if lang and has_parser(lang) then
+      return lang
+    end
+  end
+end
```

Comment:
Leverage the `return`s to reduce nesting.

```suggestion
local function resolve_lang(alias)
  if has_parser(alias) then
    return alias
  end
 
  local lang = vim.treesitter.language.get_lang(alias)
  if lang and has_parser(lang) then
    return lang
  end
end
```